addon/utils.js

```diff
@@ -121,14 +149,7 @@ export const _ComputedProperty = ComputedProperty;
  *   the task, in milliseconds
  */
 export function timeout(ms) {
-  let timerId;
-  let promise = new Promise(r => {
-    timerId = later(r, ms);
-  });
-  promise.__ec_cancel__ = () => {
-    cancel(timerId);
-  };
-  return promise;
+  return new TimeoutYieldable(ms);
```

Comment:
correct me if I'm wrong, but this change means it no longer works with async/await, right? maybe we could add a then() method that uses the yieldable method?

Reply:
the base class `Yieldable` has a `then` defined on it that will turn the yieldable into a cancelable promise, so it _should_ still work with async/await. I noticed we don't have any tests for that, so I'm going to add those once I get back to this tonight.

Reply:
🤦‍♂ sorry for the noise. you're totally right!